vm_server/send/test.py

```diff
@@ -0,0 +1,210 @@
+
+
+
+def start_server():
+  # serve(APP, host='127.0.0.1', port=5000)
+  APP.run(debug=True, host='127.0.0.1', port=5000, use_reloader=False)
+
+def execute_commands(proto_text_number):
+  """Executes commands to compile and create a proto file
+
+  Args:
+    proto_text_number: File number of the text file from where
+    the input information for the proto request is read.
+    For example, query1.txt, query2.txt etc
+  """
+  global REQUEST_ID
+  global RESPONSE
+  file_name = "query" + str(proto_text_number) + ".txt"
+  logging.debug("Executing %s test", file_name)
+  logging.debug("Creating proto file from %s", file_name)
+  os.system("python .\\proto\\create_proto.py .\\proto\\" + file_name)
+  task_request = Request_pb2.TaskRequest()
+  with open(ROOT + "input_request.pb", "rb") as input_request:
+    task_request.ParseFromString(input_request.read())
+  with open(ROOT + "input_request.pb", "rb") as input_request:
+    response = requests.post(url=URL, files={"task_request": input_request})
+    REQUEST_ID = task_request.request_id
+    task_status_response = Request_pb2.TaskStatusResponse()
+    task_status_response.ParseFromString((response.content))
+    logging.debug("Initial response : %s", str(task_status_response))
+    with open(ROOT + "initial_response.pb", "wb") as initial_response:
+      initial_response.write(task_status_response.SerializeToString())
+      initial_response.close()
+    if task_status_response.status == Request_pb2.TaskStatusResponse.ACCEPTED:
+      logging.debug("Request was accepted.")
+      logging.debug("Starting up server and listening to the response")
+      RESPONSE = False
+      # thread = threading.Thread(target=start_server)
+      thread = KThread(target=start_server)
+      thread.start()
+      thread.join(int(task_request.timeout) + 10)
+      if thread.is_alive():
+        thread.kill()
+        thread.join()
+      if RESPONSE is True:
+        logging.debug("Test %s passed", str(proto_text_number))
+      else:
+        logging.debug("Test %s failed", str(proto_text_number))
+    else:
+      logging.debug("Request was not accepted")
+      logging.debug(str(task_status_response))
+  logging.debug(type(response))
+  logging.debug(str(response.content))
+  if os.path.exists(ROOT + "response.pb"):
+    os.remove(ROOT + "response.pb")
+  with open(ROOT + "response.pb", "wb") as f:
+    f.write(response.content)
+    f.close()
+
+if __name__ == "__main__":
+  logging.basicConfig(filename="server.log",
+                      level=logging.DEBUG,
+                      format="%(asctime)s:%(levelname)s: %(message)s")
+  logging.getLogger().addHandler(logging.StreamHandler())
+  # APP.run(debug=True, host='127.0.0.1', port=5000)
+  if len(sys.argv) != 2:
+    usage_message()
+    sys.exit(-1)
+  if arguments.test_flag == ALL_TESTS:
+    for file_id in range(1, 5):
+      try:
+        execute_commands(file_id)
+      except Exception as err:
+        logging.debug(err)
```

Comment:
Should we use pyunit (https://wiki.python.org/moin/PyUnit) for testing instead? Will have a better interface than we doing the exception catching and exiting

Reply:
Agreed. Used pytest framework for testing in python instead.